app/services/create_reservation.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class CreateReservation
+  def call
+    return false unless seats_selected?
+
+    ActiveRecord::Base.transaction do
+      reservation.save!
+      create_tickets
+    rescue StandardError
+    end
+  end
+
+  private
+
+  attr_reader :reservation, :params
+
+  def initialize(reservation, params)
```

Comment:
You don't need to declare it explicitly as private - [see this](http://jackxxu.github.io/ruby/2016/06/02/ruby-private-initialize-method/).

Reply:
make sense, I moved it out form private